src/lib.rs

```diff
@@ -94,59 +113,129 @@ fn write_mod_cmd<W: Write>(config: &ModConfig, ctx: &ConfigContext, mut writer:
     Ok(())
 }
 
-fn run_mod_tasks(config: &ModConfig, ctx: &ConfigContext) -> Result<()> {
+fn run_mod_tasks(config: &ModConfig, ctx: &ConfigContext, log: &mut Logger) -> Result<()> {
+    static TASKS_DONE: Lazy<Mutex<HashSet<String>>> = Lazy::new(|| Mutex::new(HashSet::new()));
+    let cybercmd_path = Path::new(ctx.game_dir).join("tools").join("cybercmd");
+    fs::create_dir_all(&cybercmd_path).unwrap();
+    let cybercmd_path = cybercmd_path.canonicalize().unwrap();
+    let cybercmd_path = cybercmd_path.to_str().unwrap();
+    const NO_WINDOW_FLAGS: u32 = 0x08000000;
+
     for task in &config.tasks {
-        match task {
-            Task::InvokeScc {
-                path,
-                custom_cache_dir,
-                terminate_on_errors,
-            } => {
-                let cmd = Path::new(ctx.game_dir)
-                    .join("engine")
-                    .join("tools")
-                    .join("scc.exe");
-                let path = render(path, ctx.clone());
-                let custom_bundle = render(custom_cache_dir, ctx.clone());
-                const NO_WINDOW_FLAGS: u32 = 0x08000000;
-
-                let res = Command::new(&cmd)
-                    .arg("-compile")
-                    .arg(path)
-                    .arg("-customCacheDir")
-                    .arg(custom_bundle)
-                    .creation_flags(NO_WINDOW_FLAGS)
-                    .status()
-                    .ok();
-                if *terminate_on_errors == Some(true) && !matches!(res, Some(st) if st.success()) {
-                    std::process::exit(0);
+        let mut tasks_done = TASKS_DONE.lock().unwrap();
```

Comment:
I'm missing the point of the hash set and obtaining a lock for it inside the loop, are you trying to prevent this function from being executed from multiple threads and running concurrently? That shouldn't happen, but if you want to prevent that you could just lock on entering the function

Reply:
The hash set is to prevent tasks from being run multiple times. This was inspired by my own addon which has a "InvokeScc" task to work around the bug where cybercmd requires a task to be defined to parse a toml file (it wont let one with just args.)

The mutex is an artifact of me not knowing much about hooking in third party processes, and learning rust. It appeared much of the code was being executed twice, possibly from different threads. So I used once_cell, and when I still ran into issues I wrapped it in a mutex. It may just be I am failing to understand something.

Reply:
`try_get_final_cmd` should only be executed once, it's already behind a `sync::Lazy` so I don't think a mutex will make any difference
there was an issue with the hook running from multiple executables - that's why there's a `exe.eq_ignore_ascii_case("Cyberpunk2077")` check in the code, maybe that's what you were seeing, but I still don't think it should run the tasks twice

Reply:
That makes a lot of sense. Could it be redscript itself? Since the InvokeScc code doesn't change the working directory, if the ASI loader is in any dll that gets loaded into the redscript process, it would also load the asi files?

Also the Cyberpunk loader / REDLoader could do it. But I'm running Cyberpunk2077.exe directly.

Whatever the case, that would explain some of the multiple execution behaviour I was seeing. How some of the statics were being initialized more than once, despite once_cell.

Reply:
Ooo, or the error reporter executable. I can't recall if Windows loads proxies from the cwd. But it would make sense *not* to do that. That leaves one other executable in the cyberpunk folder, `REDEngineErrorReporter.exe`.

Reply:
I think it was running from some process started by the game, IIRC you can find out by logging `std::env::current_exe()`